### PR TITLE
Added a few location while parsing

### PR DESCRIPTION
**Pull request by samsa1**
Added the locations at 3 parsing locations where it had been forgotten.

This does not impact the interface because nobody reads those locations but it is cleaner to have them.

### CODE REVIEW

File: parsing/parser.mly [hunk 1]
```diff
@@ -3390,7 +3390,7 @@ label_declaration_semi:
   attrs2 = post_item_attributes
     { let docs = symbol_docs $sloc in
       let attrs = attrs1 @ attrs2 in
-      Te.mk tid cs ~params ~priv ~attrs ~docs,
+      Te.mk tid cs ~params ~priv ~attrs ~docs ~loc:(make_loc $sloc),
```
Comment (gasche) [inline parsing/parser.mly]: `let loc = make_loc $sloc in` would be more consistent with other cases above.

**Review by gasche — APPROVED** [review]
Fine, thanks!

File: parsing/parser.mly (hunk 1 above)
Comment (samsa1) [reply parsing/parser.mly]: It is indeed more coherent with the other changes but not with the overall cases in the lines around that one (for example on line 3416)

Comment (samsa1) [reply parsing/parser.mly]: Should I do this change to have coherence with similar cases or keep it that way for local coherence ?

Comment (gasche) [reply parsing/parser.mly]: Whatever. I think `let loc` is slightly nicer, but Those Who Do the Work Decide, so let's keep things as you prefer.

File: Changes [hunk 2]
```diff
@@ -129,6 +129,9 @@ Working version
 
 ### Internal/compiler-libs changes:
 
+- #XXXXX: Added location to exception definitions and type extensions
+  (Samuel Vivien, review by ...)
```
Comment (gasche) [inline Changes]: You should complete this and the PR is good to go.

File: parsing/parser.mly (hunk 1 above)
Comment (samsa1) [reply parsing/parser.mly]: I concluded that it was indeed nicer if we didn't care about the local coherence and did the modification

File: Changes (hunk 2 above)
Comment (samsa1) [reply Changes]: Should be good now

**Pull request merged**